inference-engine/src/inference_engine/src/tensor.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/runtime/tensor.hpp"
+
+#include "blob_factory.hpp"
+#include "ie_ngraph_utils.hpp"
+
+namespace ov {
+namespace runtime {
+
+Tensor::Tensor(const std::shared_ptr<void>& so,
+               const ov::element::Type type,
+               const ov::PartialShape shape,
+               const std::shared_ptr<ie::Blob>& impl)
+    : _so{so},
+      _type{type},
+      _shape{shape},
+      _impl{impl} {
+    if (_impl == nullptr) {
+        IE_THROW() << "Tensor was not initialized.";
+    }
+}
+
+Tensor::Tensor(const ov::element::Type type, const ov::PartialShape shape) : _type{type}, _shape{shape} {
+    ie::SizeVector ie_shape;
+    for (auto&& dim : shape) {
+        ie_shape.push_back(dim.get_length());
+    }
+    _impl = make_blob_with_precision({convertPrecision(_type), ie_shape, ie::Layout::NCHW});
+    _impl->allocate();
+}
+
+Tensor::Tensor(const ov::element::Type type, const ov::PartialShape shape, void* ptr) : _type{type}, _shape{shape} {
+    ie::SizeVector ie_shape;
+    for (auto&& dim : shape) {
+        ie_shape.push_back(dim.get_length());
+    }
+    _impl = make_blob_with_precision({convertPrecision(_type), ie_shape, ie::Layout::NCHW}, ptr);
+}
+
+ov::element::Type Tensor::get_element_type() const {
+    if (_impl == nullptr) {
+        IE_THROW(NotAllocated) << "Tensor was not initialized.";
+    }
+    return _type;
+}
+ov::PartialShape Tensor::get_shape() const {
+    if (_impl == nullptr) {
+        IE_THROW(NotAllocated) << "Tensor was not initialized.";
+    }
+    return _shape;
+}
+
+void* Tensor::data(const element::Type type) const {
+    if (_impl == nullptr) {
+        IE_THROW(NotAllocated) << "Tensor was not initialized.";
+    }
+    if (type != element::undefined) {
+        if (type != _type) {
+            IE_THROW(ParameterMismatch) << "Tensor element type should be: " << _type << " casted to " << type;
+        }
+    }
+    try {
+        return InferenceEngine::as<InferenceEngine::MemoryBlob>(_impl)->rmap().as<void*>();
```

Comment:
we need to make sure that `offset` in case of ROI is added here.

Reply:
done